src/Entity/CatalogPromotionGroup.php

```diff
@@ -65,25 +65,28 @@ public function addProduct(ProductInterface $product): void
         if (!$this->hasProduct($product)) {
             $this->products->add($product);
 
-            $product->setAppliedCatalogPromotionGroup($this);
+            $product->setCatalogPromotionGroup($this);
```

Comment:
@dujesaric do you need to have this line, if you add and persist line before, maybe it will be added without this line?